internal/pkg/aws/s3/s3.go

```diff
@@ -52,3 +60,46 @@ func (s *Service) PutArtifact(bucket, fileName string, data io.Reader) (string,
 
 	return resp.Location, nil
 }
+
+// EmptyBucket deletes all objects within the bucket.
+func (s *Service) EmptyBucket(bucket string) error {
+	var listResp *s3.ListObjectVersionsOutput
+	var err error
+	listParams := &s3.ListObjectVersionsInput{
+		Bucket: aws.String(bucket),
+	}
+	// Remove all versions of all objects.
+	for {
+		listResp, err = s.s3Svc.ListObjectVersions(listParams)
+		if err != nil {
+			return fmt.Errorf("list objects for bucket %s: %w", bucket, err)
+		}
+		var objectsToDelete []*s3.ObjectIdentifier
+		for _, object := range listResp.Versions {
+			objectsToDelete = append(objectsToDelete, &s3.ObjectIdentifier{
+				Key:       object.Key,
+				VersionId: object.VersionId,
+			})
+		}
+		// After deleting other versions, remove delete markers version.
+		// For info on "delete marker": https://docs.aws.amazon.com/AmazonS3/latest/dev/DeleteMarker.html
+		for _, deleteMarker := range listResp.DeleteMarkers {
+			objectsToDelete = append(objectsToDelete, &s3.ObjectIdentifier{
+				Key:       deleteMarker.Key,
+				VersionId: deleteMarker.VersionId,
+			})
+		}
+		_, err = s.s3Svc.DeleteObjects(&s3.DeleteObjectsInput{
+			Bucket: aws.String(bucket),
+			Delete: &s3.Delete{
+				Objects: objectsToDelete,
```

Comment:
`listResp.DeleteMarkers` + `listResp.Versions` can not be more than 1000 objects right?

Reply:
Yeah but `objectsToDelete` shall not exceed 1000 either, because the `list` return values limit already includes both `version` and `delete marker`

Reply:
ok perfect!